### ISSUE TITLE
Just a few problems after implement - overall it is really good 😊

### ISSUE THREAD

**Issue by jalsco** [vuquangtrong/simplify-theme#4]
Hello, am very impressed with the Simplify theme. It works well and was easy to setup (for somebody with limited skills) . All links  are working already on my internal test-site  and Search, Tags, analytics etc .. are all working. All that is very good - and was so very fast to get a solid clean bootstrap website up and running. 

Thank you ! Just a few problems - hope you can help 😊

1.  I cannot get the **tag cloud** to show up ? tag_cloud plugin is installed and loaded.  And the settings in `pelicanconf` are in place. I notice that  `pelican-plugins` calls it  ` pelican.plugins.tag_cloud`  so it is loaded as that ... but when doing the `pip` install it has a different name (not with underline)  ` pip install pelican-tag-cloud`  Anyway - no tag cloud shows up on any page. If you have any tips to make it show up, those would be greatly appreciated 😊 

2.  On right side of your Article [https://www.codeinsideout.com/simplify-theme/blog/simplify-article.html](url)  **you show the TOC which highlights each item as the reader goes down the page.** That is really cool and  I would like to show that on my site also. But nothing shows up for that ... How to make it show up ? Does it need some special code entered into `pelicanconf` ?

**Comment by jalsco** [comment]
About the tag-cloud.. I do have site-stats plug-in enabled and working. I know it is working because the “5 mins” as reading time estimate does show up in the top section of each article...

**Comment by vuquangtrong** [comment]
Hi @jalsco

I have a quick look at your described problems. As I haven't used Pelican for a long time, so it'll take time to re-install and investigate. 

At this time, I think it is due to changes in Pelican engines and plugins system, started from [version 4.5](https://blog.getpelican.com/pelican-4.5-released.html), please look at the section _Upgrading from previous releases_.

I will be back to answer your problems when I have time to check them more.

Thanks for your reporting.

**Comment by jalsco** [comment]
Thank you so much for your very fast reply !  I will look at that section about upgrading. I actually only installed for first time a few days ago so I installed directly on the latest Pelican (4.7) , but I will check that section anyway.

**Comment by jalsco** [comment]
UPDATE - `tag-cloud`  **is working BUT**  it the only page it shows on is  `http://127.0.0.7:8000/tags.html`  (ie the tags page),  at the top of the page above the expandable list. 
![image](https://user-images.githubusercontent.com/7985459/142762816-7063e04c-033b-4304-9177-974a7e4cc135.png)
It seems like the problem for (1 and 2 above) is that **the sidebar** never shows...  The sidebar , on right side of article, is where I hope to see the TOC (in articles) and also where I hope to see the Tag-Cloud (for the home screen) . Seems like the problem maybe is just - **how to enable the sidebar?**

**Comment by jalsco** [comment]
Ok - those problems are now 99% resolved. Cause seems to be incompatibility with other (not related) plugins - like `extract-toc`.  I removed `extract-toc` . I now have the TOC showing in the side-bar of the Article page - just like in your example web-page

However - there is still **one** very annoying little problem with it.  It relates to the `nav-link-active class`. That is the blue highlighting of the TOC headings as the user scrolls, showing the user's current location in the page.. The problem is that - when first opening the article - the `nav-link-active` does not operate.  BUT then if I resize the webbrowser it starts to operate. It seems that the resizing must touch or start-in <1024 size... 

My very limited knowledge of CSS is already exhausted looking at this.  Can you please have a look at it ? 🙏🏻🙏🏻😊  

Gif below shows the problem - resizing the browser and the live-link hightlight comes active, but upon article open it is not active.

![toc-hightlight](https://user-images.githubusercontent.com/7985459/143183331-824b1a6a-ced4-43b2-b268-9010fdd66a3a.gif)

**Comment by vuquangtrong** [comment]
Hi @jalsco 

I updated the source code to use new plugin system. Note that I don't have much time to replace `tipue_search` with new [pelican search](https://github.com/pelican-plugins/search).

When I testing the highlighting in TOC panel, I haven't seen the problem that you reported.

![site-preview](https://user-images.githubusercontent.com/843536/143219848-50a0848b-0a70-45ba-94fc-f9c40879ae52.GIF)

How did you test your site? I use  both `pelican --listen` in `build` folder, and directly open the generated `.html` files in `docs` folder, they all work find with highlighting.

The piece of code that handles adding "active" class to ToC item is in the `bootstrap.min.js` file. You can download a newer one from [get Bootstrap](https://getbootstrap.com/docs/5.1/getting-started/download/) under _Compiled CSS and JS_ section.

**Comment by jalsco** [comment]
Thanks for your quick reply. I have since done further testing...  

That problem in gif above shows like that only on Firefox, not Chrome - for my site. But when I look at your site the problem does not show on either Firefox or Chrome.  My conclusion is the problem in in `bootstrap.min.js` and is not caused by the theme.  

I will try with your suggestion:  

> The piece of code that handles adding "active" class to ToC item is in the bootstrap.min.js file. You can download a newer one from get Bootstrap under Compiled CSS and JS section.

About `tipue_search` - it is working perfectly for me. I did try to modify the theme myself to replace it with the new Pelican search, but it was too complex for me to do and is unnecessary for me - as `tipue_search` is working perfectly. 

I test my site using `make clean` `make html` and `make serve` , then viewing it on port 8000. 

I consider the problem has been fully investigated and effectively solved. Thanks again 😊👍🏻

**Comment by jalsco** [comment]
Postscript: I followed your suggestion and replaced:

`/simplify-theme/static/bootstrap/bootstrap.min.js `
`/simplify-theme/static/bootstrap/bootstrap.min.css `

with downloaded files from `bootstrap-5.1.3-dist`

The results are interesting....  It made it worse - like this:  With the 5.1.3 files **both Chrome and Firefox**  showed the same problem as described above, with original (min.js & min.css) only Firefox showed that problem case. Also - it is actually a little worse than before because no amount of window-resizing activated the blue highlighting of the TOC headings as the user scrolls - in either Browser. .  

I changed those 2 files back to the original and what I described above returned, that is: Chrome (all good), Firefox (upon article loading the highlighting of TOC is not working, but after some window resizing it begins to work).  

Chrome:  96.0.4664.45 (Official Build) (64-bit)
Firefox: 94.0.2 (64-bit)
